src/realm/sync/client.cpp

```diff
@@ -754,7 +754,18 @@ bool SessionImpl::process_flx_bootstrap_message(const SyncProgress& progress, Do
     }
 
     bool new_batch = false;
-    bootstrap_store->add_batch(query_version, std::move(maybe_progress), received_changesets, &new_batch);
+    try {
+        bootstrap_store->add_batch(query_version, std::move(maybe_progress), received_changesets, &new_batch);
+    }
+    catch (const LogicError& ex) {
+        if (ex.kind() == LogicError::binary_too_big) {
+            IntegrationException ex(ClientError::bad_changeset,
```

Comment:
You can even throw `bad_changeset_size` if you wanted it since there is both a ClientError and ProtocolError and we can translate it before sending to the server.

Reply:
Done!